paasta_tools/firewall.py

```diff
@@ -50,48 +57,98 @@ def chain_name(self):
+        rules.extend(_smartstack_rules(conf, self.soa_dir))
+        return tuple(rules)
+
+    def update_rules(self):
+        iptables.ensure_chain(self.chain_name, self.rules)
+
+
+def _default_rule(conf):
+    policy = conf.get_outbound_firewall()
+    if policy == 'block':
+        return iptables.Rule(
+            protocol='ip',
+            src='0.0.0.0/0.0.0.0',
+            dst='0.0.0.0/0.0.0.0',
+            target='REJECT',
+            matches=(),
+        )
+    elif policy == 'monitor':
+        # TODO: log-prefix
+        return iptables.Rule(
+            protocol='ip',
+            src='0.0.0.0/0.0.0.0',
+            dst='0.0.0.0/0.0.0.0',
+            target='LOG',
+            matches=(),
+        )
+    else:
+        raise AssertionError(policy)
+
+
+def _well_known_rules(conf):
+    for resource in conf.get_dependencies().get('well-known', ()):
+        if resource == 'internet':
+            yield iptables.Rule(
                 protocol='ip',
                 src='0.0.0.0/0.0.0.0',
                 dst='0.0.0.0/0.0.0.0',
-                target='REJECT',
+                target='PAASTA-INTERNET',
                 matches=(),
-            ),
-            iptables.Rule(
-                protocol='tcp',
-                src='0.0.0.0/0.0.0.0',
-                dst='169.254.255.254/255.255.255.255',
-                target='ACCEPT',
-                matches=(
-                    ('tcp', (('dport', '20668'),)),
-                )
-            ),
+            )
+        else:
+            # TODO: handle better
+            raise KeyError(resource)
+
+
+def _smartstack_rules(conf, soa_dir):
+    for namespace in conf.get_dependencies().get('smartstack', ()):
+        # TODO: handle non-synapse-haproxy services
+        # TODO: support wildcards?
+        service, _ = namespace.split('.', 1)
+        service_namespaces = get_all_namespaces_for_service(service, soa_dir=soa_dir)
+        port = dict(service_namespaces)[namespace]['proxy_port']
+
+        yield iptables.Rule(
+            protocol='tcp',
+            src='0.0.0.0/0.0.0.0',
+            dst='169.254.255.254/255.255.255.255',
+            target='ACCEPT',
+            matches=(
+                ('tcp', (('dport', six.text_type(port)),)),
+            )
         )
 
-    def update_rules(self):
-        iptables.ensure_chain(self.chain_name, self.rules)
 
-
-def active_service_groups():
+def active_service_groups(soa_dir):
     """Return active service groups."""
-    # TODO: actually read these from somewhere
-    return {
-        ServiceGroup('cool_service', 'main', 'block'): {
+    from paasta_tools import firewall_update
+    service_groups = collections.defaultdict(set)
+    for service, instance, framework in firewall_update.services_running_here(soa_dir):
+        service_groups[ServiceGroup(service, instance, framework, soa_dir)].add(
+            # TODO: how to get the mac address of this container?
             '02:42:a9:fe:00:02',
```

Comment:
Oh wait I see some with paasta_service and paasta_instance. format_docker_parameters() does it